Snowflake.Data.Tests/SFConnectionPoolT.cs

```diff
@@ -504,6 +541,98 @@ public void TestConcurrentConnectionPoolingAsync()
+                connection.BeginTransaction();
+                Assert.Throws<SnowflakeDbException>(() =>
+                {
+                    using (var command = connection.CreateCommand())
+                    {
+                        command.CommandText = "invalid command will throw exception and leave session with an unfinished transaction";
+                        command.ExecuteNonQuery();
+                    }
+                });
+            }
+
+            using (var connectionWithSessionReused = new SnowflakeDbConnection())
+            {
+                connectionWithSessionReused.ConnectionString = ConnectionString;
+                connectionWithSessionReused.Open();
+                
+                Assert.AreEqual(firstOpenedSessionId, connectionWithSessionReused.SfSession.sessionId);
+                using (var cmd = connectionWithSessionReused.CreateCommand())
+                {
+                    cmd.CommandText = "SELECT CURRENT_TRANSACTION()";
+                    Assert.AreEqual(DBNull.Value, cmd.ExecuteScalar());
+                    Assert.AreEqual(false, connectionWithSessionReused.HasActiveTransaction());
+                }
+            } 
+            
+            Assert.AreEqual(1, SnowflakeDbConnectionPool.GetCurrentPoolSize(), "Connection should be reused and any pending transaction rolled back before it gets back to the pool");
+        }
+
+        [Test]
+        public void TestRollbackTransactionOnPooledWhenConnectionClose()
+        {
+            SnowflakeDbConnectionPool.SetMaxPoolSize(1);
+
+            string firstOpenedSessionId = null;
+            using (var connection1 = new SnowflakeDbConnection())
+            {
+                connection1.ConnectionString = ConnectionString;
+                connection1.Open();
+                using (var command = connection1.CreateCommand())
+                {
+                    firstOpenedSessionId = connection1.SfSession.sessionId;
+                    command.CommandText = "BEGIN TRANSACTION";
+                    command.ExecuteNonQuery();
+                    Assert.AreEqual(true, connection1.HasActiveTransaction());
+                }
+            }
+
+            using (var connection2 = new SnowflakeDbConnection())
+            {
+                connection2.ConnectionString = ConnectionString;
+                connection2.Open();
+                using (var command = connection2.CreateCommand())
+                {
+                    Assert.AreEqual(firstOpenedSessionId, connection2.SfSession.sessionId);
+                    command.CommandText = "SELECT CURRENT_TRANSACTION()";
+                    Assert.AreEqual(DBNull.Value, command.ExecuteScalar());
+                    Assert.AreEqual(false, connection2.HasActiveTransaction());
+                }
+            }
+
+            Assert.AreEqual(1, SnowflakeDbConnectionPool.GetCurrentPoolSize(), "Connection should be returned to the pool");
```

Comment:
the second connection came back to pool? what about first connection that has active transaction? in line ~601 we should have some assertion of pool size?

Reply:
added new asserts